### ISSUE TITLE
IGMPv3 not working

### ISSUE THREAD

**Issue by greenpau** [troglobit/pimd#48]
Configured all interfaces to use IGPMv3 via procfs. e.g. 

```
echo 3 > /proc/sys/net/ipv4/conf/all/force_igmp_version
```

note: i changed this value on all interfaces.

On the wire:

```
[root@ip-192-168-16-66 ~]# tshark -i tun1759 -f "igmp"
Running as user "root" and group "root". This could be dangerous.
Capturing on 'tun1759'
  1   0.000000 10.253.17.59 -> 224.0.0.1    IGMPv2 48 Membership Query, general
1   2   3.642623 10.253.17.59 -> 224.0.0.13   IGMPv2 48 Membership Report group 224.0.0.13
2   3   6.770621 10.253.17.59 -> 224.0.0.22   IGMPv2 48 Membership Report group 224.0.0.22
3   4   9.074621 10.253.17.59 -> 224.0.0.2    IGMPv2 48 Membership Report group 224.0.0.2
4
```

The issues is that I am still using IGMPv2. Any ideas how to force IGMPv3?

**Comment by troglobit** [comment]
Running pimd/mrouted means starting a multicast router. In UNIX kernels this means userspace takes on the role of sending IGMP/MLD queries.  The pimd master currently only supports IGMPv2, but in pull request #36 (which I've still not completed my audit of) IGMPv3 is included.

I'm expecting to merge PR #36 within a few weeks, it's simply been too much to do at work (and life) currently.

**Comment by greenpau** [comment]
@troglobit , thank you for the response! :smile: Will wait for the merge.

**Comment by troglobit** [comment]
:smiley: :+1:

**Comment by troglobit** [comment]
Support for IGMPv3 merged for the upcoming v2.3.0 release.  It is now the default on all `phyint` interfaces (use new `igmpv2` option to a `phyint` to force v2 instead).

I have tested it with Linux (Ubuntu 15.04), OpenBSD 5.7 and NetBSD 6.1.5, only ASM though, not SSM.
